common/src/main/java/org/apache/celeborn/common/haclient/MasterNotLeaderException.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.haclient;
+
+import java.io.IOException;
+
+// This class is reserved for compatible with 0.2 client
+// TODO remove it in 0.4.0
```

Comment:
Does this mean the PR target is branch-0.3?

Reply:
Yes, this should be included in 0.3 to gain compatibility with 0.2 clients.

Reply:
Then the 0.4 server does not gain compatibility with 0.2 clients? What compatibility strategy should we provide?

Reply:
Before 1.0
- 0.2 client support 0.2 and 0.3 server
- 0.3 client support 0.3 and 0.4 server
- ...

Reply:
thank you for clarifying.

then change the PR target branch from `main` to `branch-0.3`?

Reply:
changed

Reply:
> Before 1.0
> 
> * 0.2 client support 0.2 and 0.3 server
> * 0.3 client support 0.3 and 0.4 server
> * ...

+1